src/Apps/ArtAdvisor/05-Near-Object-Rail/server.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  createReference,
+  getNearObjects,
+  getObjects,
+  getObjectWithId,
+} from "Apps/ArtAdvisor/05-Near-Object-Rail/helpers/weaviate"
+import chalk from "chalk"
+import express, { Request, Response } from "express"
+import { filter } from "lodash"
+
+const likeArtwork = async (req: Request, res: Response) => {
+  const { artworkId, userId } = req.body
+
+  try {
+    await createReference({
+      referenceProperty: "likedArtworks",
+      fromClassName: "DiscoveryUsers",
+      fromWeaviateId: userId,
+      toClassName: "DiscoveryArtworks",
+      toWeaviateId: artworkId,
+    })
+
+    res.end()
+  } catch (e) {
+    console.error(chalk.red("ERROR: "), e)
+    res.status(500).send("An error occurred")
+  }
+}
+
+const dislikeArtwork = async (req: Request, res: Response) => {
+  const { artworkId, userId } = req.body
+
+  try {
+    await createReference({
+      referenceProperty: "dislikedArtworks",
+      fromClassName: "DiscoveryUsers",
+      fromWeaviateId: userId,
+      toClassName: "DiscoveryArtworks",
+      toWeaviateId: artworkId,
+    })
+
+    res.end()
+  } catch (e) {
+    console.error(chalk.red("ERROR: "), e)
+    res.status(500).send("An error occurred")
+  }
+}
+
+const nearObject = async (req: Request, res: Response) => {
+  console.log("nearObject")
+
+  try {
+    const userId = req.params.id
+
+    const userQuery =
+      "name likedArtworks { ... on DiscoveryArtworks { title _additional { id }}} dislikedArtworks { ... on DiscoveryArtworks { title _additional { id }}} _additional { id }"
+
+    const weaviateUser = await getObjectWithId({
+      objectId: userId,
+      query: userQuery,
+      className: "DiscoveryUsers",
+    })
+
+    const userLikes =
+      weaviateUser?.data?.Get?.DiscoveryUsers[0].likedArtworks || []
+    const userDislikes =
+      weaviateUser?.data?.Get?.DiscoveryUsers[0].dislikedArtworks || []
+
+    const artworksQuery = "title imageUrl _additional { id }"
+
+    let artworks
+
+    if (userLikes.length > 0) {
+      artworks = await getNearObjects({
+        objectId: userId,
+        objectClassName: "DiscoveryUsers",
+        targetClassName: "DiscoveryArtworks",
+        query: artworksQuery,
+        limit: 50,
```

Comment:
The results of this search are calculated solely on the artworks "liked". We only track "disliked" artworks to filter them from the results. This is a trade off with the `nearObect` [search](https://github.com/artsy/force/pull/13982/files#r1636854765) vs the `nearText` search, which [allows us to](https://weaviate.io/developers/weaviate/api/graphql/search-operators#variables-2) `moveAwayFrom` and object.